config/k8s/wait.py

```diff
@@ -130,44 +136,53 @@ def main(args):
     # Find pod where job ran
     #
     # Reload credentials to reduce risk of token expiration
-    config.load_kube_config(context=args.context, persist_config=False)    
+    config.load_kube_config(context=args.context, persist_config=False)
     core = client.CoreV1Api()
     pod = find_pod(core, args)
     if pod is None:
         print("ERROR: Failed to find pod")
         sys.exit(1)
-    if len(pod.items) < 1:
+    pods_num = len(pod.items)
+    if pods_num < 1:
         print("WARNING: no pods found with job name {}".format(args.job_name))
         sys.exit(0)
-    if len(pod.items) > 1:
+    # Do not quit job even if more than 1 pod found, one might be successfull
+    # Check last one
+    if pods_num > 1:
         print("WARNING: >1 pod found with job name {}".format(args.job_name))
-        sys.exit(0)
-    pod_name = pod.items[0].metadata.name
-    print("Found job on pod {}".format(pod_name))
-
-    #
-    # Get logs (from pod)
-    #
 
-    # default: check the main container logs
-    cont_name = pod.items[0].spec.containers[0].name
-
-    # unless the initContainer failed, get that log, because if the
-    # initContainer failed, the main container will not have run
-    if (pod.items[0].spec.init_containers):
-        init_cont_name = pod.items[0].spec.init_containers[0].name
-        if not pod.items[0].status.init_container_statuses[0].ready:
-            print("ERROR: initContainer {} not ready / failed.".format(init_cont_name))
-            cont_name = init_cont_name
-            k8s_success = False
-    try:
-        log = core.read_namespaced_pod_log(name=pod_name,
-                                           namespace=args.namespace,
-                                           container=cont_name)
-        print("Container Log:")
-        print(log)
-    except client.rest.ApiException as e:
-        print("Exception: Unable to get pod log", e)
+    log = ""
+    # Try to collect logs from all pods created by this job
+    for jpod in pod.items:
+        pod_name = jpod.metadata.name
+        print("Found job on pod {}".format(pod_name))
+
+        #
+        # Get logs (from pod)
+        #
+
+        # default: check the main container logs
+        cont_name = jpod.spec.containers[0].name
+
+        # unless the initContainer failed, get that log, because if the
+        # initContainer failed, the main container will not have run
+        if (jpod.spec.init_containers):
```

Comment:
No need for parenthesis.

```suggestion
        if jpod.spec.init_containers:
```